### PR TITLE
Allow for pluggable external commands on PATH

### PR DESCRIPTION
**Pull request by meatballhat**
plus support for "run commands"/"rc file"-style config

### CODE REVIEW

File: tfw [hunk 1]
```diff
@@ -59,9 +55,27 @@ main() {
...
+  done
+
   local command="${1}"
   shift || true
-  if "__run_${command:-help}" "${@}"; then
+  if command -v "__run_${command:-help}" &>/dev/null &&
+    "__run_${command:-help}" "${@}"; then
+    exit 0
+  elif command -v "tfw-${command}" &>/dev/null &&
+    "tfw-${command}" "${@}"; then
```
Comment (soulshake) [inline tfw]: I feel like I've run into problems before using hyphens in commands, but I can't reproduce now, so oh well! :woman_shrugging:

Comment (soulshake) [reply tfw]: Also, maybe I'm reading this wrong, but aren't we losing error output if the command fails here? Or are we trusting the command to report?

Comment (meatballhat) [reply tfw]: Sounds fun! :smile_cat:

File: tfw [hunk 2]
```diff
@@ -151,6 +165,23 @@ Administrivia:
   admin-users
 
 EOUSAGE
+
+  shopt -s nullglob
+  local external_commands=()
+  for path_prefix in ${PATH//:/ }; do
```
Comment (soulshake) [inline tfw]: Note to self: remember this `${PATH//:/ }` usefulness!

File: tfw [hunk 3]
```diff
@@ -151,6 +165,23 @@ Administrivia:
...
+    for exe in "${path_prefix}"/tfw-*; do
+      external_commands=("${external_commands[@]}" "${exe##*tfw-}")
```
Comment (soulshake) [inline tfw]: What does `"${exe##*tfw-}")` do?

File: tfw (hunk 2 above)
Comment (meatballhat) [reply tfw]: It's handy, but definitely brittle if one's `PATH` entries include spaces :frowning_face: .... but I'm feeling lazy I guess?

File: tfw (hunk 3 above)
Comment (meatballhat) [reply tfw]: Strip `*tfw-` from the front of the string, equivalent to `string.gsub(/^.*tfw-/, '')`

File: tfw (hunk 1 above)
Comment (meatballhat) [reply tfw]: We shouldn't be losing error output for the command execution, which happens on line 78.  Line 77 is asking a question where the exit code is all that matters to me.  Does that help answer your question?

**Pull request merged**